projects/openapi-utilities/src/utilities/json-changelog.ts

```diff
@@ -144,62 +78,110 @@ type JsonChangelog = {
 };
 
 export function jsonChangelog(
-  groupedChanges: ReturnType<typeof groupChangesAndRules>
+  specs: { from: OpenAPIV3.Document; to: OpenAPIV3.Document },
+  groupedChanges: GroupedDiffs
 ): JsonChangelog {
   const results: JsonChangelog = { operations: [] };
 
-  const { changesByEndpoint, specification } = groupedChanges;
-  for (const [_, endpointChange] of changesByEndpoint) {
-    results.operations.push(getEndpointLogs(endpointChange));
+  const { paths } = groupedChanges;
+  for (const [path, pathObject] of Object.entries(paths)) {
```

Comment:
This felt a little insane to do - i ended up moving the path logic into the groupDiffs function so we don't need to inteprete what "adding a path" means https://github.com/opticdev/optic/pull/1505/commits/6995a8d9939ea7a146dd887629d97717e2a00d59